smarts/env/multi_scenario_env.py

```diff
@@ -344,6 +354,8 @@ def step(
 
         limited_actions: Dict[str, np.ndarray] = {}
         for agent_name, agent_action in action.items():
+            if not agent_name in self._prev_obs:
+                continue
```

Comment:
What is the purpose of these two lines?

Reply:
There were cases where the agent was not in the previous observation. There was no way to then constrain the agent's action because there needs to be at least one history frame.

Reply:
I think actions were only expected for agents with observation. These two lines here might allow errors to slip by unnoticed. Anyway, I think we should simply remove this older `multi_scenario_env.py` since we now have `RelativeTargetPose` action space.